openshift_tools/monitoring/generic_metric_sender.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# vim: expandtab:tabstop=4:shiftwidth=4
+"""
+Base class for sending metrics
+"""
+
+import os
+import yaml
+
+
+class GenericMetricSenderException(Exception):
+    '''
+        GenericMetricSenderException
+        Exists to propagate errors up from the api
+    '''
+    pass
+
+
+class GenericMetricSender(object):
+    """
+    Abstract class that encapsulates all metric sending operations.
+    """
+
+    def __init__(self):
+        self.config = None
+        self.config_file = None
+
+    def parse_config(self):
+        """ parse default config file """
+        if not self.config:
+            if not os.path.exists(self.config_file):
+                raise GenericMetricSenderException(self.config_file + " does not exist.")
+            self.config = yaml.load(file(self.config_file))
+
+    def add_dynamic_metric(self, discovery_key, macro_string, macro_array, host=None, synthetic=False):
+        pass
```

Comment:
Actually I want this implemented as "Do-Nothing" in the parent class so in the case of add_dynamic_metric for example which is not implemented in HawkSender, function will default to parent. This will allow generic calls to MetricSender no matter what concrete sender supports.
I will however, raise an exception in the constructor, to prevent instantiation of this abstract class. Thanks.